src/main/kotlin/snyk/oss/annotator/OSSMavenAnnotator.kt

```diff
@@ -0,0 +1,37 @@
+package snyk.oss.annotator
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+import snyk.oss.Vulnerability
+
+class OSSMavenAnnotator : OSSBaseAnnotator() {
+
+    override fun lineMatches(psiFile: PsiFile, line: String, vulnerability: Vulnerability) =
+        psiFile.virtualFile.name == "pom.xml" &&
+            line.contains("<artifactId>${getIntroducingPackage(vulnerability)}</artifactId>")
+
+    override fun getIntroducingPackage(vulnerability: Vulnerability): String {
+        return super.getIntroducingPackage(vulnerability).split(":")[1].replace("\"", "")
+    }
+
+    override fun textRange(psiFile: PsiFile, vulnerability: Vulnerability): TextRange {
+        return fixRange(psiFile, vulnerability)
+    }
+
+    override fun fixRange(psiFile: PsiFile, vulnerability: Vulnerability): TextRange {
+        val textRange = super.textRange(psiFile, vulnerability)
+        val text = psiFile.viewProvider.document?.text!!
```

Comment:
null case proper handling needed due to: `Can be null for binary or files without events enabled`